core/setup.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+import os
+import shutil
+import subprocess
+import codecs
+import os.path
+
+from setuptools import setup, find_packages, find_namespace_packages
+
+cwd = os.path.dirname(os.path.abspath(__file__))
+
+with open(os.path.join('..', 'VERSION')) as version_file:
+    version = version_file.read().strip()
+
+"""
+To release a new stable version on PyPi, simply tag the release on github, and the Github CI will automatically publish 
+a new stable version to PyPi using the configurations in .github/workflows/pypi_release.yml . 
+You need to increase the version number after stable release, so that the nightly pypi can work properly.
+"""
+try:
+    if not os.getenv('RELEASE'):
+        from datetime import date
+        today = date.today()
+        day = today.strftime("b%Y%m%d")
+        version += day
+except Exception:
+    pass
+
+def create_version_file():
+    global version, cwd
+    print('-- Building version ' + version)
+    version_path = os.path.join(cwd, 'src', 'autogluon', 'core', 'version.py')
+    with open(version_path, 'w') as f:
+        f.write('"""This is autogluon version file."""\n')
+        f.write("__version__ = '{}'\n".format(version))
+
+
+long_description = open(os.path.join('..', 'README.md')).read()
+
+MIN_PYTHON_VERSION = '>=3.6.*'
+
+requirements = [
+    'Pillow<=6.2.1',
+    'numpy>=1.16.0',
+    'scipy>=1.3.3',
+    'cython',
+    'tornado>=5.0.1',
+    'requests',
+    'matplotlib',
+    'tqdm>=4.38.0',
+    'paramiko>=2.4',
+    'dask>=2.6.0',
+    'cryptography>=2.8',
+    'distributed>=2.6.0',
+    'ConfigSpace<=0.4.10',
+    'graphviz<0.9.0,>=0.8.1',
```

Comment:
how these `<` are decided, just blocking to use a future major release? if another library installed by the user requires a higher version, then there is a conflict.

Reply:
unsure why the graphviz < was decided, but most tabular specific < are just to avoid new major releases that may have API breaking changes. We limit sklearn to minor version due to past experiences of it breaking API in minor version update.